src/protocol/attribution/mod.rs

```diff
@@ -0,0 +1,60 @@
+use crate::secret_sharing::Replicated;
+
+mod accumulate_credit;
+
+#[allow(clippy::module_name_repetitions)]
+#[derive(Debug, Copy, Clone)]
+pub struct AttributionInputRow<F> {
+    is_trigger_bit: Replicated<F>,
+    helper_bit: Replicated<F>,
+    #[allow(dead_code)]
+    breakdown_key: Replicated<F>,
+    value: Replicated<F>,
+}
+
+pub struct AccumulateCreditInputRow<F> {
+    stop_bit: Replicated<F>,
+    credit: Replicated<F>,
+    report: AttributionInputRow<F>,
+}
+
+#[allow(dead_code)]
+pub struct AccumulateCreditOutputRow<F> {
+    breakdown_key: Replicated<F>,
+    credit: Replicated<F>,
+    aggregation_bit: Replicated<F>,
+}
+
+struct IterStep {
+    name: &'static str,
+    count: u32,
+    id: String,
+}
+
+impl IterStep {
+    pub fn new(name: &'static str) -> Self {
+        Self {
+            name,
+            count: 0,
+            id: String::from(name),
+        }
+    }
+
+    fn next(&mut self) -> &Self {
+        self.count += 1;
+        self.id = format!("{}_{}", self.name, self.count);
```

Comment:
there will be $O(N)$ allocations here, we probably want to mark this as todo to fix. I think @martinthomson was planning to add a cheap way to create a unique step id for release builds

Reply:
This should go away once we decide to move away from `AsRef<str>`. I understand AsRef<str> for steps is temporary for this prototype?